lib/spawn.ts

```diff
@@ -1,8 +1,57 @@
 import { Ok } from "./result.ts";
-import type { ScopeInternal } from "./scope.ts";
+import type { ScopeInternal } from "./scope-internal.ts";
 import { createTask, type NewTask } from "./task.ts";
 import type { Effect, Operation, Task } from "./types.ts";
 
+/**
+ * Run another operation concurrently as a child of the current one.
+ *
+ * The spawned operation will begin executing immediately and control will
+ * return to the caller when it reaches its first suspend
+ *
+ * @example
+ * ```js
+ * import { main, sleep, suspend, spawn } from 'effection';
+ *
+ * await main(function*() {
+ *   yield* spawn(function*() {
+ *     yield* sleep(1000);
+ *     console.log("hello");
+ *   });
+ *   yield* spawn(function*() {
+ *     yield* sleep(2000);
+ *     console.log("world");
+ *   });
+ *   yield* suspend();
+ * });
+ * ```
+ *
+ * You should prefer using the spawn operation over calling
+ * {@link Scope.run} from within Effection code. The reason being that a
+ * synchronous failure in the spawned operation will not be caught
+ * until the next yield point when using `run`, which results in lines
+ * being executed that should not.
+ *
+ * @example
+ * ```js
+ * import { main, suspend, spawn, useScope } from 'effection';
+ *
+ * await main(function*() {
+ *   yield* useScope();
+ *
+ *   scope.run(function*() {
+ *    throw new Error('boom!');
+ *   });
+ *
+ *   console.log('this code will run and probably should not');
+ *
+ *   yield* suspend(); // <- error is thrown after this.
+ * });
+ * ```
+ * @param operation the operation to run as a child of the current task
+ * @typeParam T the type that the spawned task evaluates to
```

Comment:
This should say `@typeParam {T}`

Reply:
Should it always be the very first word following `@typeParam`?

Reply:
After discussing in discord, we're opted to try identical handling to `@param` which uses the first word following the annotation followed by an  `"-"`